kotlinx-coroutines-core/common/src/flow/Builders.kt

```diff
@@ -277,31 +283,34 @@ public fun <T> channelFlow(@BuilderInference block: suspend ProducerScope<T>.()
  * fun flowFrom(api: CallbackBasedApi): Flow<T> = callbackFlow {
  *     val callback = object : Callback { // implementation of some callback interface
  *         override fun onNextValue(value: T) {
- *             // Note: offer drops value when buffer is full
- *             // Use either buffer(Channel.CONFLATED) or buffer(Channel.UNLIMITED) to avoid overfill
- *             offer(value)
+ *             // To avoid blocking you can configure channel capacity using
+ *             // either buffer(Channel.CONFLATED) or buffer(Channel.UNLIMITED) to avoid overfill
+ *             try {
+ *                 sendBlocking(value)
+ *             } catch (e: Exception) {
+ *                 // Handle exception from the channel: failure in flow or premature closing
+ *             }
  *         }
  *         override fun onApiError(cause: Throwable) {
  *             cancel(CancellationException("API Error", cause))
  *         }
  *         override fun onCompleted() = channel.close()
  *     }
  *     api.register(callback)
- *     // Suspend until either onCompleted or external cancellation are invoked
+ *     /*
+ *      * Suspends until either 'onCompleted' from the callback is invoked
+ *      * or flow collector is cancelled (e.g. by 'take(1)' or because a collector's activity was destroyed).
+ *      * In both cases, callback will be properly unregistered.
+ *      */
```

Comment:
A few notes on this comment:
* "Suspends until either 'onCompleted' from the callback is invoked" -> should mention `onApiError`, too
* "or because a collector's activity was destroyed" looks out-of-place here -> replace with "collector's coroutine was cancelled"